### PR TITLE
Turing SDK: submit batch ensembling job

### PR DESCRIPTION
**Pull request by romanwozniak**
### Context:
This PR extends `turing-sdk` with a functionality of configuring and submitting batch ensembling jobs. 

### Changes:
- re-generated REST client from openapi spec (so, changes to [sdk/turing/generated](https://github.com/romanwozniak/turing/tree/sdk_submit_job/sdk/turing/generated) can be safely skipped during review)
- added necessary classes related to batch ensembling under [sdk/turing/batch](https://github.com/romanwozniak/turing/tree/sdk_submit_job/sdk/turing/batch)
- added e2e example on how to submit a job at [sdk/samples/batch_job](https://github.com/romanwozniak/turing/tree/sdk_submit_job/sdk/samples/batch_job)
- some minor changes to [proto](https://github.com/romanwozniak/turing/blob/sdk_submit_job/engines/batch-ensembler/api/proto/v1/batch_ensembling_job.proto) definition and [openapi](https://github.com/romanwozniak/turing/blob/sdk_submit_job/api/openapi/jobs.yaml) spec to make them compatible with each other

### TODO
 - [x] add unit tests where missed

### CODE REVIEW

File: sdk/turing/batch/config/source.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+    def __init__(self, dataset, join_on, columns):
+        super(EnsemblingJobPredictionSource, self).__init__(dataset, join_on)
+        self._columns = columns
+
+    @property
+    def columns(self) -> Iterable[str]:
+        return self._columns
+
+    def to_open_api(self) -> OpenApiModel:
```
Comment (krithika369) [inline sdk/turing/batch/config/source.py:58]: In general, is it right to say that the contents in `sdk/turing/batch/...` act as a simple wrapper over the generated code, for the users to interact with, and the `to_open_api` and `from_open_api` allow for interfacing with the Turing API through the generated schema?

**Review by krithika369 — APPROVED** [review]
LGTM!

File: sdk/tests/batch/config/config_test.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
+import pytest
+import turing.batch.config
+import turing.generated.models
+
+
+@pytest.mark.parametrize(
```
Comment (ashwinath) [inline sdk/tests/batch/config/config_test.py:6]: kind of a nitpick but thats one long decorator that is hard to see at first glance

File: sdk/turing/batch/job.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    JobBuildingImage --▶ JobFailedBuildImage
+        |
+        |
+        |
+        |
+        ▼
+    JobRunning --▶ JobFailed
+        |
+        |
+        |--▶ JobTerminated
```
Comment (ashwinath) [inline sdk/turing/batch/job.py:27]: There's a JobTerminating process before this.

**Review by ashwinath — APPROVED** [review]

File: api/openapi/jobs.yaml [hunk 4]
```diff
@@ -58,14 +34,39 @@ paths:
             type: array
             items:
               $ref: "#/components/schemas/EnsemblerJobStatus"
+          style: form
+          explode: true
```
[the remark this answers is not in the document]
Comment (romanwozniak) [reply api/openapi/jobs.yaml:38]: I like the level of attention you put into your reviews 🙏
Yes, `style=form + explode=true` is default for OAS3, but [openapi-generator](https://github.com/OpenAPITools/openapi-generator) doesn't recognize it and by default serializes it as CSV, unless `style` and `explode` is set explicitly

File: sdk/turing/batch/config/source.py (hunk 1 above)
Comment (romanwozniak) [reply sdk/turing/batch/config/source.py:58]: Yes, correct. This wrapper is (hopefully) more user friendly and allows us to better control the user flow

File: sdk/turing/_base_types.py [hunk 5]
```diff
@@ -28,18 +54,10 @@ def id(self, id: int):
     def created_at(self) -> datetime:
         return self._created_at
 
-    @created_at.setter
```
[the remark this answers is not in the document]
Comment (romanwozniak) [reply sdk/turing/_base_types.py:31]: No needed. Removed

File: sdk/turing/_base_types.py [hunk 6]
```diff
@@ -2,7 +2,33 @@
 from turing.generated.model_utils import ModelNormal
 
 
-class ApiObject:
+class DataObject:
+    def to_dict(self):
```
[the remark this answers is not in the document]
Comment (romanwozniak) [reply sdk/turing/_base_types.py:6]: There is no real need for `Dataset` to inherit it. But it's for pretty-printing of these classes that inherit it, that can be useful for debugging

File: sdk/tests/batch/config/config_test.py (hunk 2 above)
Comment (romanwozniak) [reply sdk/tests/batch/config/config_test.py:6]: What's the suggestion for improvement? :)

File: sdk/turing/batch/job.py (hunk 3 above)
Comment (romanwozniak) [reply sdk/turing/batch/job.py:27]: Right

**Pull request merged**